src/Swift.Runtime/src/Swift/Runtime/ProtocolConformanceDescriptor.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+
+namespace Swift.Runtime;
+
+public readonly struct ProtocolConformanceDescriptor : IEquatable<ProtocolConformanceDescriptor>
+{
+    private readonly IntPtr _handle;
+
+    private ProtocolConformanceDescriptor(IntPtr handle)
+    {
+        _handle = handle;
+    }
+
+    public readonly static ProtocolConformanceDescriptor Zero = default;
+
+    public bool IsValid => _handle != IntPtr.Zero;
+
+    public bool Equals(ProtocolConformanceDescriptor other)
+    {
+        return _handle == other._handle;
+    }
+
+    internal static bool TryGetProtocolConformanceDescriptor<TType, TProtocol>([NotNullWhen(true)] out ProtocolConformanceDescriptor? result)
+        where TProtocol : class
+    {
+        var type = typeof(TType);
+
+        if (typeof(ISwiftObject).IsAssignableFrom(type))
+        {
+            var helperType = typeof(ProtocolConformanceDescriptorHelper<,>).MakeGenericType(typeof(TType), typeof(TProtocol));
+            result = (ProtocolConformanceDescriptor)helperType.GetMethod("GetProtocolConformanceDescriptor")!.Invoke(null, null)!;
```

Comment:
Are we sure that `GetMethod("GetProtocolConformanceDescriptor")!` always returns non-null method?

Reply:
I cannot imagine a scenario this fails. We define the `GetProtocolConformanceDescriptor` function ourselves in another file